components/external_storage/src/export.rs

```diff
@@ -209,17 +221,30 @@ impl<S: ExternalStorage> ExternalStorage for AutoEncryptLocalRestoredFileExterna
         )
         .await
     }
+
+    fn iter_prefix(&self, prefix: &str) -> LocalBoxStream<'_, io::Result<BlobObject>> {
+        self.storage.iter_prefix(prefix)
+    }
+
+    fn delete(&self, name: &str) -> LocalBoxFuture<'_, io::Result<()>> {
+        self.storage.delete(name)
+    }
 }
 
 #[async_trait]
-impl<Blob: BlobStorage> ExternalStorage for BlobStore<Blob> {
+impl<Blob: BlobStorage + IterableStorage + DeletableStorage> ExternalStorage for Compat<Blob> {
```

Comment:
What about that only expose BlobStorage for `component/backup` and `component/importer`? Maybe we can make sure there is no iterate or delete operation in them.

Reply:
We probably cannot export a structure to an exact crate. But we have scheduled remove `BlobStorage`, so perhaps we can tidy this up at that time.